src/test_c2s_srv.c

```diff
@@ -280,9 +285,8 @@ int test_c2s(int ctlsockfd, tcp_stat_agent* agent, TestOptions* testOptions,
         log_println(0, "C2S test error: can't create pipe.");
       } else {
         if ((c2s_childpid = fork()) == 0) {
-          /* close(ctlsockfd); */
           close(testOptions->c2ssockfd);
-          close(recvsfd);
+          close(c2s_conn.socket);
```

Comment:
Pulling forward comment from previous rounds
>mtlynch replied 3 days ago
>Is there any reason to not replace all close(*.socket) with close_connection?

>pboothe replied 3 days ago
>I put SSL_shutdown in close_connection. It should have been there since the beginning.

>close(.socket), when * is a Connection, should be a close_connection. close(testOptions->c2ssockfd); is not such a case.
  
>mtlynch replied 3 days ago
>Sorry, that was confusing. I wasn't referring to that line. I was just noticing that in general we have a lot of instances where we call close on a Connection and it sounds like you're saying those should all be close_connection:

>src/test_sfw_srv.c:378:      close(sfw_conn.socket);
>src/test_sfw_srv.c:390:      close(sfw_conn.socket);
>src/test_sfw_srv.c:409:      close(sfw_conn.socket);
>src/test_sfw_srv.c:421:      close(sfw_conn.socket);
>src/test_sfw_srv.c:432:    close(sfw_conn.socket);
>src/test_s2c_srv.c:332:            close(s2c_conn.socket);
>src/test_c2s_srv.c:289:          close(c2s_conn.socket);